src/lambda/local/configureLocalLambda.ts

```diff
@@ -131,31 +132,66 @@ function getTabSize(editor?: vscode.TextEditor): number {
 
 async function loadSymbols(
     uri: vscode.Uri,
-    maxAttempts = 3,
-    retryDelayMillis = 500
-): Promise<vscode.DocumentSymbol[] | undefined> {
-    if (maxAttempts <= 0) {
-        return undefined
-    }
+    maxAttempts = 10,
+    retryDelayMillis = 200
+): Promise<vscode.DocumentSymbol[] | boolean> {
 
     const symbols = await vscode.commands.executeCommand<vscode.DocumentSymbol[]>(
         'vscode.executeDocumentSymbolProvider',
         uri
     )
+    // checking if symbols exists as this can fail if the VS Code JSON symbol provider is not yet initialized
     if (symbols) {
+        // file has symbols if JSON with at least one valid top-level key/value pair
         return symbols
     }
 
+    if (maxAttempts <= 0) {
```

Comment:
This method's responsibility went from "get symbols from a file" to "get symbols from a file but if it can't then ask the user what they want to do", meaning an additional concern was added. This also makes the return value less clear ("array OR boolean with specific meaning). 

What if we leave the return value as it was ("array or undefined") and make this prompting logic the responsibility of the caller? The function remains a "symbol retriever" and our caller can decide whether or not it wants to call into the method again, and the caller can also decide if it will clobber the file.